internal/oauth2/lazy_token_test.go

```diff
@@ -0,0 +1,80 @@
+package oauth2
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestLazyToken(t *testing.T) {
+	nextTokenID := 0
+	mockServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Could we check here that the request being sent is correct? That it has the correct grant_type, client_id, and client_secret set? If this is already happening somewhere else in a test that's ok too, I just think it's worth checking that the request the tokener sends is what we expect.